compression/gzip_zinfo.c

```diff
@@ -341,83 +398,75 @@ int extract_data_from_buffer(void* d, offset_t datalen, struct gzip_zinfo* index
     return ret;
 }
 
+int extract_data_from_file(const char* file, struct gzip_zinfo* index, offset_t offset, void* buf, int len) {
+    FILE* fp = fopen(file, "rb");
+    if (fp == NULL) 
+        return GZIP_ZINFO_FILE_NOT_FOUND;
+
+    int ret = extract_data_from_fp(fp, index, offset, buf, len);
+    fclose(fp);
+    return ret;
+}
 
-int extract_data_fp(FILE *in, struct gzip_zinfo *index, offset_t offset, void *buffer, int len)
-{
+// This is the same as extract_data_fp, but instead of a file, it decompresses
+// data from a buffer which contains the exact data to decompress
+int extract_data_from_buffer(void *d, offset_t datalen,
```

Comment:
This function doesn't have logic change other than reformat and changing to `init_flate` helper